internal/cmd/skupper/listener/kube/listener_create.go

```diff
@@ -0,0 +1,224 @@
+	cmd.CobraCmd.Flags().StringVarP(&cmd.flags.listenerType, "type", "t", "tcp", "The listener type. Choices: [tcp].")
+	cmd.CobraCmd.Flags().StringVarP(&cmd.flags.output, "output", "o", "", "print resources to the console instead of submitting them to the Skupper controller. Choices: json, yaml")
+}
+
+func (cmd *CmdListenerCreate) ValidateInput(args []string) []error {
+	var validationErrors []error
+	resourceStringValidator := validator.NewResourceStringValidator()
+	numberValidator := validator.NewNumberValidator()
+	listenerTypeValidator := validator.NewOptionValidator(utils.ListenerTypes)
+	outputTypeValidator := validator.NewOptionValidator(utils.OutputTypes)
+
+	// Validate arguments name and port
+	if len(args) < 2 {
+		validationErrors = append(validationErrors, fmt.Errorf("listener name and port must be configured"))
+	} else if len(args) > 2 {
+		validationErrors = append(validationErrors, fmt.Errorf("only two arguments are allowed for this command"))
+	} else if args[0] == "" {
+		validationErrors = append(validationErrors, fmt.Errorf("listener name must not be empty"))
+	} else if args[1] == "" {
+		validationErrors = append(validationErrors, fmt.Errorf("listener port must not be empty"))
+	} else {
+		ok, err := resourceStringValidator.Evaluate(args[0])
+		if !ok {
+			validationErrors = append(validationErrors, fmt.Errorf("listener name is not valid: %s", err))
+		} else {
+			cmd.name = args[0]
+		}
+
+		cmd.port, err = strconv.Atoi(args[1])
+		if err != nil {
+			validationErrors = append(validationErrors, fmt.Errorf("listener port is not valid: %s", err))
+		}
+		ok, err = numberValidator.Evaluate(cmd.port)
+		if !ok {
+			validationErrors = append(validationErrors, fmt.Errorf("listener port is not valid: %s", err))
+		}
+	}
+
+	// Validate there is already a site defined in the namespace before a listener can be created
+	siteList, _ := cmd.client.Sites(cmd.namespace).List(context.TODO(), metav1.ListOptions{})
+	if siteList == nil || len(siteList.Items) < 1 {
+		validationErrors = append(validationErrors, fmt.Errorf("A site must exist in namespace %s before a listener can be created", cmd.namespace))
+	}
+
+	// Validate if there is already a listener with this name in the namespace
+	if cmd.name != "" {
+		listener, err := cmd.client.Listeners(cmd.namespace).Get(context.TODO(), cmd.name, metav1.GetOptions{})
+		if listener != nil && !errors.IsNotFound(err) && listener.Status.StatusMessage == "Ok" {
+			validationErrors = append(validationErrors, fmt.Errorf("there is already a listener %s created for namespace %s", cmd.name, cmd.namespace))
+		}
+	}
+
+	// Validate flags
+	if cmd.flags.routingKey != "" {
+		ok, err := resourceStringValidator.Evaluate(cmd.flags.routingKey)
+		if !ok {
+			validationErrors = append(validationErrors, fmt.Errorf("routing key is not valid: %s", err))
+		}
+	}
+	if cmd.flags.host != "" {
+		//TBD what characters are not allowed
+	}
```

Comment:
Sorry for focusing on the details! The comment is absolutely correct, we can leave it there (//TBD what characters are not allowed in the host) and remove the if statement, because that it has nothing inside.

Reply:
removed if statement and left comment so we can come back and add check later.

Reply:
I believe this commit is missing

Reply:
sorry didn't push the fix.   Done now